src/kOS/Screen/TermWindow.cs

```diff
@@ -444,10 +444,14 @@ void ProcessKeyEvents()
             Event e = Event.current;
             if (e.type == EventType.KeyDown)
             {
-                // Unity handles some keys in a particular way
-                // e.g. Keypad7 is mapped to 0xffb7 instead of 0x37
-                var c = (char)(e.character & 0x007f);
-
+                // This ugly hack is here to solve a bug with Unity mapping
+                // Keycodes to Unicode chars incorrectly on its Linux version:
+                char c;
+                if ((e.character & 0xff00) == 0xff00) // Only trigger on Unicode values 0xff00 through 0xffff, to avoid issue #2061
+                    c = (char)(e.character & 0x007f); // When doing this to solve issue #206
+                else
+                    c = e.character;
+                
```

Comment:
This silly workaround we had to put in a few years ago to work around a Unity on Linux bug, caused multi-byte Unicode to break now.

### Reason for the original workaround code:

When mapping a vanilla ASCII character into Unicode, you're supposed to pad the upper bits with zero.  So for example, 0x41 ("A") should become 0x0041.  But on their Linux port, for some reason, while Unity did this correctly for printable ASCII chars, they did not do so for control ASCII chars (i.e. (0x01 (control-A), 0x08 (control-H, or 'tab'), etc).  For them, Unity padded them with 1's instead of with 0's.  So 0x0d (control-M, or "return key"), instead of giving a unicode value of 0x000d like it should, was giving a value of 0xff0d.  This made the return key fail to work in kOS on Linux.  So at the time I added this ugly hack to 'fix' it by simply masking off all the high bits.

But this broke reading keys that send non-ascii Unicode chars, because the high byte was being *always* clobbered no matter what, whether it was the problem case in the Linux port (0x000d becoming 0xff0d) or not.

### The new version

Now it only clobbers the high byte when it's *exactly* all 1's, which should never occur for any normal printable character, as the Unicode from 0xff00 through 0xffff is considered "private use" that the Unicode consortium will never assign to any standard meaning.